cmd/soroban-rpc/internal/db/db.go

```diff
@@ -265,3 +311,22 @@ func runMigrations(db *sql.DB, dialect string) error {
 	_, err := migrate.ExecMax(db, dialect, m, migrate.Up, 0)
 	return err
 }
+
+// NewTestDB is here so that other packages in sRPC can use it
```

Comment:
The db functionality should be unit tested within this package. For example, the code in cmd/soroban-rpc/internal/db/transaction.go should be unit tested here instead of in  cmd/soroban-rpc/internal/methods/get_transaction_test.go . In cmd/soroban-rpc/internal/methods/get_transaction_test.go , you can use mocks for the transaction store. And in cmd/soroban-rpc/internal/test/transaction_test.go we have the integration tests which exercises everything end to end with real implementations

Reply:
@Shaptic I think this comment still applies. I think we should move `NewTestDB()` back to a testing file in this package

Reply:
Ah yes I forgot about this.

>  In cmd/soroban-rpc/internal/methods/get_transaction_test.go , you can use mocks for the transaction store.

In the [current tests](https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/methods/get_transaction_test.go#L154), you can actually see that this does not happen: we use a real store. That's why when I transitioned the tests, I did the same thing. I saw that we have this weird imitation of a fake backend for [get_latest_ledger_test.go](https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/methods/get_latest_ledger_test.go#L76) but I couldn't understand what that was testing tbh.

Anyway, I added a fake in-memory store for use in testing in 5e49d7d and refactored the methods to use it. I think that's sufficient!